### PR TITLE
Rwc2017 challenge set a table

### PR DESCRIPTION
**Pull request by Rayman**

### CODE REVIEW

File: challenge_set_a_table/src/challenge_set_a_table_states/config.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+MIN_GRAB_OBJECT_HEIGHT = 0.075
+MAX_GRAB_OBJECT_WIDTH = 0.15
```
Comment (reinzor) [inline challenge_set_a_table/src/challenge_set_a_table_states/config.py]: Please put this in the robocup_knowledge

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 2]
```diff
@@ -22,8 +28,15 @@ def fetch(robot, time_out=15.0):
     i = 0
     while not confirmed and i < 10:
```
Comment (reinzor) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: why 10?

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 3]
```diff
@@ -33,54 +46,168 @@ def fetch(robot, time_out=15.0):
...
+    """ Gets the breakfast order by asking it to the customer. N.B.: the order is asked, but this information is not
+     stored (we won't get points for that anyway """
+    def __init__(self, robot, options, grasp_designator1, grasp_designator2, grasp_designator3, timeout=15.0):
+        """ Constructor
+
+        :param robot: robot object
+        :param options: list with strings containing the possible breakfast choices
+        :param timeout: timeout in seconds
+        """
+        smach.State.__init__(self, outcomes=["done"])
```
Comment (reinzor) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: To knowledge file?

File: challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+import smach
+
+# TU/e
+import robot_skills
+import robot_smach_states as states
+import robot_smach_states.util.designators as ds
+
+# Challenge set the table
+# from entity_description_designator import EntityDescriptionDesignator
+from config import MIN_GRAB_OBJECT_HEIGHT, MAX_GRAB_OBJECT_WIDTH
```
Comment (reinzor) [inline challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py]: to knowledge

File: challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+        # Try to place the object
+        item = ds.EdEntityDesignator(robot=self._robot, id=arm.occupied_by.id)
+        arm_designator = ds.ArmDesignator(all_arms={arm.side: arm}, preferred_arm=arm)
+        sm = states.Place(robot=self._robot, item_to_place=item, place_pose=self._place_designator, arm=arm_designator)
+        result = sm.execute()
+
+        # If failed, do handover to human in order to continue
+        if result != "done":
+            sm = states.HandoverToHuman(robot=self._robot, arm_designator=arm_designator)
+            sm.execute()
```
Comment (reinzor) [inline challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py]: Dynamically creating a state machine is not nice

Comment (reinzor) [reply challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py]: @jlunenburg

File: challenge_set_a_table/src/challenge_set_a_table_states/config.py (hunk 1 above)
Comment (jlunenburg) [reply challenge_set_a_table/src/challenge_set_a_table_states/config.py]: That would mean duplicating it

File: challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py (hunk 5 above)
Comment (jlunenburg) [reply challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py]: Point taken but takes too much time to implement

File: challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+                break
+
+        if arm is None:
+            return "failed"
+
+        # Try to place the object
+        item = ds.EdEntityDesignator(robot=self._robot, id=arm.occupied_by.id)
+        arm_designator = ds.ArmDesignator(all_arms={arm.side: arm}, preferred_arm=arm)
+        sm = states.Place(robot=self._robot, item_to_place=item, place_pose=self._place_designator, arm=arm_designator)
+        result = sm.execute()
```
Comment (LoyVanBeek) [inline challenge_set_a_table/src/challenge_set_a_table_states/manipulate_machine.py]: Nope, please use the proper nesting methods

**Comment by jlunenburg** [comment]
I created issue #409 to accomodate the state machine hacks and addressed the other comments. @reinzor and @LoyVanBeek satisfied? If so, please approve ;)

**Pull request merged**